UtopiaCity/Controllers/Sport/RequestToAdministrationController.cs

```diff
@@ -0,0 +1,188 @@
+            var request = _mapper.Map<RequestToAdministration>(requestViewModel);
+            _requestToAdministrationService.AddRequestToDb(request);
+            return RedirectToAction(nameof(AllRequestsToAdministration));
+        }
+
+        [HttpGet]
+        public IActionResult Delete(string id)
+        {
+            if (id == null)
+            {
+                return View("Error", "The id is incorrect. Please, try again");
+            }
+
+            RequestToAdministration request = _requestToAdministrationService.GetRequestToAdministrationById(id);
+            if (request == null)
+            {
+                return View("Error", "The request is not found. Please, try again");
+            }
+
+            var requestViewModel = _mapper.Map<RequestToAdministrationViewModel>(request);
+            return View(requestViewModel);
+        }
+
+        [HttpPost, ActionName("Delete")]
+        public IActionResult DeleteConfirmed(string id)
+        {
+            if (id == null)
+            {
+                return View("Error", "The id is incorrect. Please, try again");
+            }
+
+            RequestToAdministration request = _requestToAdministrationService.GetRequestToAdministrationById(id);
+            if (request == null)
+            {
+                return View("Error", "The request is not found. Please, try again");
+            }
+
+            _requestToAdministrationService.RemoveRequestFromDb(request);
+            return RedirectToAction(nameof(AllRequestsToAdministration));
+        }
+
+        [HttpGet]
+        public IActionResult Edit(string id)
+        {
+            if (id == null)
+            {
+                return View("Error", "The id is incorrect. Please, try again");
+            }
+
+            RequestToAdministration request = _requestToAdministrationService.GetRequestToAdministrationById(id);
+            if (request == null)
+            {
+                return View("Error", "The request is not found. Please, try again");
+            }
+
+            var requestViewModel = _mapper.Map<RequestToAdministrationViewModel>(request);
+            return View(requestViewModel);
+        }
+
+        [HttpPost]
+        public IActionResult Edit(string id, RequestToAdministrationViewModel requestViewModel)
+        {
+            if (id == null)
+            {
+                return View("Error", "The id is incorrect. Please, try again");
+            }
+            else if (requestViewModel == null)
+            {
+                return View("Error", "Some errors in input data. Please, try again");
+            }
+
+            SportComplex sportComplex = _sportComplexService.GetSportComplexByTitle(requestViewModel.SportComplexTitle);
+            if (sportComplex == null)
+            {
+                return View("Error", "Some problems with the sport complex data. Please, try again");
+            }
+
+            if (requestViewModel.IsApproved && !sportComplex.Available)
```

Comment:
probably sportcomplex.available can be ignored and just overwritten based on requestViewModel. You are updating db anyway, so overwriting bool property should not be a problem really